### DISCUSSION TITLE
Any news on support for MacOs

### DISCUSSION THREAD

**Discussion by kadst43** [o3de/o3de#11357] [General]
Hi all,

As a professor teaching classes on game development, I'd like to use O3DE for some of my classroom projects.  In particular, I'm going to be teaching a Game AI class in the spring that I think it could be great for... except that many of my students only have access to Macs.  I see on the website that "Support for developing on macOS is in an experimental stage. Documentation coming in 2022..."  Any word on when in 2022?

**Comment by galibzon** [comment]
@kadst43, thanks for your interests in O3DE. Officially, MacOS is still under experimental support. Meaning, some engineers use it internally but there are no guarantees that all features work as in Windows & Linux.

**Comment by Windvis** [comment]
I'm also interested in a macOS version. Is there an overview of things that prevents macOS from being officially supported? I understand that it's too early to make official stable releases but would it be possible to get macOS nightly builds, similar to the other platforms?

**Reply by rgba16f** [reply]
There isn't an official doc that lists what it would take to promote Mac to officially supported.

From my personal local tests the main thing that is broken on Mac (vs PC) is user mouse input, trying to control the camera in the editor will quickly show unexpected behavior. As far as I know this is an error in our QT layer. The main impediment to that getting fixed is available engineering time.

There are also other polish issues to do with the AssetProcessor, packaging the build into an App bundle, etc. These are not blockers to working on the platform but they do reduce the quality of the experience.

I would recommend taking your question about nightly Mac builds to [sig-platform](https://discord.gg/5ktSVU4R) and/or [sig-build](https://discord.gg/qezeSw9d) on discord.

**Comment by Python-Ninja-Hebi** [comment]
Is a version available on macOs?

**Reply by shdwshard** [reply]
There are no prebuilds for MacOS, so you'll need to check out the source and build it yourself.  Adding to the trouble is that both main and development produce unusable mac binaries at the moment.
I'm bisecting to find the latest useful version at the moment, but try ecda84240ade77e8a4f2f886cde5e45bcf31ec88. That seems to work well.

**Reply by shdwshard** [reply]
@Python-Ninja-Hebi Through bisecting, I found 4ac8adff is the most recent which will function on my M1 Mac. You might try that as a starting point, along with the following build commands:
```bash
# Configure the editor's build
cmake -B ./build/mac_xcode -S . -G Xcode -DLY_ASSET_DEPLOY_MODE=LOOSE -DLY_ASSET_DEPLOY_ASSET_TYPE=mac
# Build the editor
cmake --build build/mac_xcode --target Editor  --config profile
# Run the editor
./build/mac_xcode/bin/profile/Editor.app/Contents/MacOS/Editor
```

**Reply by Anthony-Gaudino** [reply]
Also, for anyone trying this, you should install Rosetta by running `softwareupdate --install-rosetta` and also install Cmake: https://formulae.brew.sh/formula/cmake#default

-----

https://github.com/o3de/o3de/commit/4ac8adffbe54068a43b7e90d3248b7cf309ee84 worked for me on M1 Mac, but a very recent version failed while building the code.

With https://github.com/o3de/o3de/commit/4ac8adffbe54068a43b7e90d3248b7cf309ee84 I created a small project and none of the FBX assets build, so I had no 3D assets in editor.
The editor also crashes very often.

**Comment by ccmvn** [comment]
Can someone provide a beta mac version?

**Comment by elolelo** [comment]
Has anyone successfully managed to build the MacOS version for themselves?

**Comment by anonymousssus** [comment]
can anyone pre-build O3DE.dmg or O3DE.app/O3DE Launcher.app

**Comment by anonymousssus** [comment]
(I cannot build it by myself, it is too time-consuming.)

**Comment by adrian-antoci** [comment]
Hi everyone, any more news on this? Lots of us are on M1 laptops and really can't wait to try it.
